### PR TITLE
Use auto for basic types

### PR DESCRIPTION
**Pull request by fire**
Work towards converting auto to basic types for readability in online review tools.

### CODE REVIEW

File: src/sandbox.cpp [hunk 1]
```diff
@@ -478,9 +478,9 @@ Variant &Sandbox::get_mutable_scoped_variant(unsigned index) {
 		ERR_PRINT("Invalid scoped variant index.");
 		throw std::runtime_error("Invalid scoped variant index.");
 	}
-	auto *var = state().scoped_variants[index - 1];
+	const godot::Variant *var = state().scoped_variants[index - 1];
 	// Find the variant in the variants list
-	auto it = std::find_if(state().variants.begin(), state().variants.end(), [var](const Variant &v) {
+	std::__wrap_iter<godot::Variant *> it = std::find_if(state().variants.begin(), state().variants.end(), [var](const Variant &v) {
```
Comment (fwsGonzo) [inline src/sandbox.cpp:483]: This needs to be reverted

File: src/sandbox.cpp [hunk 2]
```diff
@@ -629,7 +629,7 @@ bool Sandbox::get_property(const StringName &name, Variant &r_ret) {
 }
 
 const SandboxProperty *Sandbox::find_property_or_null(const StringName &name) const {
-	for (auto &prop : m_properties) {
+	for (SandboxProperty &prop : m_properties) {
```
Comment (fwsGonzo) [inline src/sandbox.cpp:632]: const

File: src/sandbox_functions.cpp [hunk 3]
```diff
@@ -1163,7 +1163,7 @@ PackedStringArray Sandbox::get_functions() const {
 	PackedStringArray result;
 	// Get all unmangled public functions from the guest program.
 	// Exclude functions that belong to the C/C++ runtime, as well as compiler-generated functions.
-	for (auto &function : machine().memory.all_unmangled_function_symbols()) {
+	for (std::string_view &function : machine().memory.all_unmangled_function_symbols()) {
```
Comment (fwsGonzo) [inline src/sandbox_functions.cpp:1166]: std::string_view by reference is almost always a bug, I think we need to have a look at this. If they are string_views, iterating it by value would be preferrable

File: src/sandbox_functions.cpp [hunk 4]
```diff
@@ -1214,7 +1214,7 @@ Sandbox::BinaryInfo Sandbox::get_program_info_from_binary(const PackedByteArray
 
 		// Get all unmangled public functions from the guest program.
 		// Exclude functions that belong to the C/C++ runtime, as well as compiler-generated functions.
-		for (auto &function : machine.memory.all_unmangled_function_symbols()) {
+		for (std::string_view &function : machine.memory.all_unmangled_function_symbols()) {
```
Comment (fwsGonzo) [inline src/sandbox_functions.cpp:1217]: Another string_view by reference - check if it can be by value

File: src/syscalls.cpp [hunk 5]
```diff
@@ -388,43 +388,43 @@ APICALL(api_obj) {
...
 			// XXX: vec->free(emu.machine());
-			auto properties = obj->get_property_list();
+			TypedArray<Dictionary> properties = obj->get_property_list();
 			auto [sptr, saddr] = vec->alloc<GuestStdString>(emu.machine(), properties.size());
 			for (size_t i = 0; i < properties.size(); i++) {
 				Dictionary dict = properties[i].operator godot::Dictionary();
-				auto name = String(dict["name"]).utf8();
-				const auto self = saddr + sizeof(GuestStdString) * i;
+				CharString name = String(dict["name"]).utf8();
+				const std::tuple_element<1, std::tuple<GuestStdString *, unsigned long long>>::type self = saddr + sizeof(GuestStdString) * i;
```
Comment (fwsGonzo) [inline src/syscalls.cpp:408]: Needs to be reverted

File: src/sandbox_functions.cpp (hunk 3 above)
Comment (fire) [reply src/sandbox_functions.cpp:1166]:
```
src/sandbox.cpp:569:4: error: 'this' argument to member function 'set' has type 'const SandboxProperty', but function is not marked const
                        prop.set(*this, value);
                        ^~~~
src/vmproperty.h:34:7: note: 'set' declared here
        void set(Sandbox &sandbox, const Variant &value);
             ^
1 error generated.
```

Comment (fire) [reply src/sandbox_functions.cpp:1166]: Can you put a commit on top mentioning how you'll fix it?

File: src/sandbox_functions.cpp (hunk 4 above)
Comment (fire) [reply src/sandbox_functions.cpp:1217]: Can you put a commit on top here too.

**Pull request merged**